### ISSUE TITLE
[Awaiting bugzero checklist and c+ newdot payment][$2000] Empty screen shows for a brief moment in change task assignee modal

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#23563]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to any task report
2. Click Assignee field
3. Observe contacts list
## Expected Result: 
Contacts list shows immediately after skeleton
## Actual Result: 
Empty list shows for a brief moment before contacts list shows
Note: this doesn't happen in new task flow

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.45-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/0a10ae27-641b-4c3e-9013-7029378dede1

https://github.com/Expensify/App/assets/43996225/5edb1b0a-3d1a-48ce-b553-92ca557cce11

**Expensify/Expensify Issue URL:**
**Issue reported by:** @situchan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690261404929219

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0163bd3cb559f30da0</li>
        <li>Upwork Job ID: 1684109952842158080</li>
        <li>Last Price Increase: 2023-08-11</li>
<li>Automatic offers: </li>
<ul>
<li>tienifr | Contributor | 26196168</li>
<li>situchan | Reporter | 26196170</li>
</ul></li>
    </ul>
</details>

**Comment by kadiealexander** [comment]
Reproduced:

https://github.com/Expensify/App/assets/59587260/8322ea21-0be3-4167-8895-bc712f6146af

**Comment by meesum1214** [comment]
**Proposal**

This is because state of skeleton is becoming false before the list contacts fetched from API completely. Its state should be false immediately when response arrives from API. Like this

```
getAllContacts(body)
      .then((res) => {
        setContacts(res.contacts);
        setLoader(false);
      })
      .catch((error) => {
        setLoader(false);
        console.log(error.response.data.message);
      });
```

About me:
I have 2+ years experience in ReactJS, NextJS, Tailwind CSS, NodeJS, Express, Sequelize, and databases like Postgres, MongoDB, and MySQL. Let's build beautiful and scalable web applications together!

Portfolio:
https://offroaddesertsafari.com/
https://kidzquran.vercel.app/
http://farmacie.agronomics.pk/
https://mintaweb.vercel.app/

Feel free to reach me out.

Best Regards,
Salman Naqvi

**Comment by meesum1214** [comment]
Contributor details
Your Expensify account email: [salmannaqvi4641@gmail.com](mailto:salmannaqvi4641@gmail.com)
Upwork Profile Link: https://www.upwork.com/freelancers/salmannaqvi

**Comment by sobitneupane** [comment]
Thanks for your [proposal](https://github.com/Expensify/App/issues/23563#issuecomment-1650190651) @samh-nl. 

> updateOptions() should be called on mount: For this, _.debounce has an immediate argument that should be set to true.

If we do show `updateOptions` will be called after each keystroke which is not a wanted situation.

> Remove shouldShowOptions={didScreenTransitionEnd} so that it has its default value of true.

 I believe `didScreenTransitionEnd` was added with some intention. So, removing it completely doesn't look like a great idea. If you think `didScreenTransitionEnd` is not needed. Can you please link the PR where was it added and why it won't be required any longer.
 
 @samh-nl Whenever you update your proposal, please follow [the contributing guideline](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#propose-a-solution-for-the-job) and use the following format:
 
 ```
 ## Proposal
[Updated](link to proposal)
 ```

**Comment by sobitneupane** [comment]
Thanks for the proposal @tienifr.

> Note There're other places (TaskShareDestinationSelectorModal, SearchPage, ...) face this problem so we should to apply the above logic too

Can you please explain what problem `TaskShareDestinationSelectorModal and `SearchPage` has? I was unable to reproduce. Or I might be missing something.

@tienifr Can you also help me understand why the issue is reproducible here but not in Assignee selection in Create task flow?

**Comment by sobitneupane** [comment]
@samh-nl 
> Are we sure that it is not reproducible there? It uses the same component and I see the same behavior.

I am talking about the blank screen being displayed when we navigate to the assignee through the Task Report. We are trying to remove that blank screen.
<img width="296" alt="Screenshot 2023-07-28 at 17 34 58" src="https://github.com/Expensify/App/assets/25876548/0ce0c456-b241-4d36-a8ed-4c35cd460e3c">


But when we navigate to the assignee page through the Create Task flow (or Search Page, etc.), than we see the loader screen but not the blank. It's okay to have the loader screen as we have few delays as you mentioned on the proposal.
<img width="296" alt="Screenshot 2023-07-28 at 17 35 57" src="https://github.com/Expensify/App/assets/25876548/d9470520-9636-42b4-a835-c0a614444df7">

**Comment by sobitneupane** [comment]
@tienifr 

Thanks for the analysis. 

There is still some black area which we are unaware of on the Root Cause. Let's try to dig deeper.

**Comment by kadiealexander** [comment]
Not overdue, will consider bumping the price tomorrow if no additional proposals come through.

**Comment by kadiealexander** [comment]
Not overdue, please get your proposals in! 📣

**Comment by pradeepmdk** [comment]
### Proposal
### Please re-state the problem that we are trying to solve in this issue.
The empty screen shows for a brief moment in the change task assignee modal

### What is the root cause of that problem?
this transaction issue. In Ios when the first modal opened with transaction end fastly. 
https://github.com/Expensify/App/blob/8b236589fa5f748dc81722545478b9ef20426604/src/components/ScreenWrapper/index.js?#L38

But in the create new task assignee time it will not happen because the previous page is in the modal. so that transaction will end with some more delay.

### What changes do you think we should make in order to solve the problem?
We need to apply transitionSpec here
https://github.com/Expensify/App/blob/main/src/libs/Navigation/AppNavigator/ModalStackNavigators.js
<img width="737" alt="Screenshot 2023-08-04 at 12 29 32 PM" src="https://github.com/Expensify/App/assets/16595846/791f53f5-24d0-437e-9eaa-bab8fa6a044e">

### What alternative solutions did you explore? (Optional)
we can pass the loading statement  once the data loaded 
https://github.com/Expensify/App/blob/8b236589fa5f748dc81722545478b9ef20426604/src/pages/tasks/TaskAssigneeSelectorModal.js#L196

### Result

https://github.com/Expensify/App/assets/16595846/409a2b4d-de93-4135-bd17-8b98d892dadc

**Comment by kadiealexander** [comment]
@sobitneupane please take a look at the above when you get a chance!

**Comment by kadiealexander** [comment]
Bumped @sobitneupane [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1691467155398209).

**Comment by pradeepmdk** [comment]
@kadiealexander @sobitneupane any updates for me

**Comment by sobitneupane** [comment]
I will review the proposal and provide update by tomorrow morning NPT.

**Comment by sobitneupane** [comment]
@pradeepmdk Thanks for the [proposal](https://github.com/Expensify/App/issues/23563#issuecomment-1665115958).

Can you please elaborate the root cause of the issue?

**Comment by sobitneupane** [comment]
> I'm so appreciate if someone can give me the more detail reason why this difference. Thanks a lot

@tienifr Do you have any update after https://github.com/Expensify/App/issues/23563#issuecomment-1655100507?

**Comment by pradeepmdk** [comment]
> @pradeepmdk Thanks for the [proposal](https://github.com/Expensify/App/issues/23563#issuecomment-1665115958).
> 
> Can you please elaborate on the root cause of the issue?

https://github.com/Expensify/App/blob/58dc0c9576f8c3d3fa8924a7fec0889232c4b537/src/libs/Navigation/linkTo.js?#L62-L74

here whenever we are navigating to the screen it will be based on the screen action type will set. in our case we are from the center to the right side so `push` will happen,
push means nothing but it **` add a new route to the navigation stack`**

In the case from create a new task after entering the name of the task, it will go to the assignee selection page. but here previous is the right side modal so here one check on `L73` previous route is the right side modal means doesn't set `push` so it will same as `navigate` will trigger.
navigate means `**navigate it first tries to find an existing route with that name, and only pushes a new route if there isn't yet one on the stack.**`

so here whenever `push` happen it will be faster on transitionEnd. but in `navigate` it will take some time.

ref: https://reactnavigation.org/docs/navigating/#navigate-to-a-route-multiple-times

### result from edit assignee:

https://github.com/Expensify/App/assets/16595846/e0abcf6b-6732-4d98-bc30-f92e3b096fa0


```
 LOG  {"payload": {"name": "RightModalNavigator", "params": {"initial": true, "params": [Object], "screen": "Task_Details", "state": undefined}, "path": undefined}, "type": "NAVIGATE"} r/2940632935871850/assignee
 LOG  action.type PUSH RightModalNavigator CentralPaneNavigator
 DEBUG  [info] Navigating to route - {"path":"/r/2940632935871850/assignee"}
 DEBUG  [info] [Pusher] Unbinding event - {"eventName":"client-userIsTyping","channelName":"private-report-reportID-2940632935871850"}
 DEBUG  [info] [Pusher] Unbinding event - {"eventName":"client-userIsTyping","channelName":"private-report-reportID-2940632935871850"}
 LOG  baseOptionsList true
 LOG  before debouncedSearch 1691599365777
 LOG  after debouncedSearch 1691599365778
 LOG  transitionEnd {"data": {"closing": false}, "target": "Task_Assignee-3BFwBUs-EyS_xCMtfCTbC", "type": "transitionEnd"}
 LOG  baseOptionsList false
 LOG  callback triggered 1691599365981
 LOG  callback triggered first  1691599366010
 LOG  baseOptionsList false
 LOG  baseOptionsList false
 LOG  baseOptionsList false
 LOG  callback triggered last 1691599366514

``` 

here you can see a callback triggered  after the transition end only 


### result from new  assignee:

https://github.com/Expensify/App/assets/16595846/a9e380c3-170e-4e97-bfa5-08f809a92dae

```
 LOG  {"payload": {"name": "RightModalNavigator", "params": {"initial": true, "params": [Object], "screen": "NewTask", "state": undefined}, "path": undefined}, "type": "NAVIGATE"} new/task/assignee
 LOG  action.type NAVIGATE RightModalNavigator RightModalNavigator
 DEBUG  [info] Navigating to route - {"path":"/new/task/assignee"}
 LOG  baseOptionsList true
 LOG  before debouncedSearch 1691599484256
 LOG  after debouncedSearch 1691599484256
 LOG  callback triggered 1691599484464
 LOG  callback triggered first  1691599484491
 LOG  baseOptionsList true
 LOG  baseOptionsList true
 LOG  baseOptionsList true
 LOG  callback triggered last 1691599484564
 LOG  transitionEnd {"data": {"closing": false}, "target": "NewTask_TaskAssigneeSelector-EWtdtI3nfzkbiQD6Vu3VN", "type": "transitionEnd"}
 LOG  baseOptionsList false
```

**Comment by pradeepmdk** [comment]
@sobitneupane For more on this pr is causing this issue 
https://github.com/Expensify/App/pull/22095/files 
this debounce timing is the problem here in our case debounce we set 0 means it will work correctly.
https://github.com/Expensify/App/blob/58dc0c9576f8c3d3fa8924a7fec0889232c4b537/src/pages/tasks/TaskAssigneeSelectorModal.js?#L100


it's hard to determine the time here because currently if we have more data it will take more time. 

```javascript
 shouldShowOptions={didScreenTransitionEnd && isLoading}
``` 
we can concat these two statements for showing the data after TransitionEnd and data also should be set in the state.
this will work in all the scenarios. even if we have more number user data.

**Comment by pradeepmdk** [comment]
@sobitneupane any updates for me

**Comment by s-alves10** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Empty screen shows for a brief moment in change task assginee modal

### What is the root cause of that problem?

When navigating between different stack navigators, the new stack navigator is mounted with its child screen on the target route and emits the `transitionEnd` event. So the `transitionEnd` event is emitted immediately after the screen is mounted. But navigating inside the same stack navigator, the new screen is mounted and pushed to the navigator. The navigator emits the `transitionEnd` event after the new screen is pushed to the navigator and so the transitionEnd event is emitted a bit later after the screen is mounted.

Let's consider the issue based on this.
When a user navigates to the change task assignee modal, the app navigates from the central pane navigator to the [TaskModalStackNavigator](https://github.com/Expensify/App/blob/ed2c287daf6a1d59f4f43a9bfc84128597575730/src/libs/Navigation/AppNavigator/ModalStackNavigators.js#L180) of the right modal navigator. We're navigating different navigators and `didScreenTransitionEnd` of the [ScreenWrapper](https://github.com/Expensify/App/blob/main/src/components/ScreenWrapper/index.js) becomes `true` after the [TaskAssigneeSelectorModal](https://github.com/Expensify/App/blob/main/src/pages/tasks/TaskAssigneeSelectorModal.js) from the above fact. But the [updateOptions](https://github.com/Expensify/App/blob/ed2c287daf6a1d59f4f43a9bfc84128597575730/src/pages/tasks/TaskAssigneeSelectorModal.js#L80) method is called at least 200ms after the screen is mounted as we debounce the method call below

https://github.com/Expensify/App/blob/ed2c287daf6a1d59f4f43a9bfc84128597575730/src/pages/tasks/TaskAssigneeSelectorModal.js#L99-L105

So `sections` calculated [here](https://github.com/Expensify/App/blob/ed2c287daf6a1d59f4f43a9bfc84128597575730/src/pages/tasks/TaskAssigneeSelectorModal.js#L118) becomes empty array for a moment and that's why we can see the empty screen.
This is the root cause of the issue

_**We can't see the empty screen when we navigate to the assignee modal through the task creation flow?**_
In this case, we navigates inside the same navigator [NewTaskModalStackNavigator](https://github.com/Expensify/App/blob/ed2c287daf6a1d59f4f43a9bfc84128597575730/src/libs/Navigation/AppNavigator/ModalStackNavigators.js#L254C7-L254C33) of the right modal navigator. According to the above fact, `transitionEnd` event is emitted after the mounted screen is pushed to the navigator. During the push action, [TaskAssigneeSelectorModal](https://github.com/Expensify/App/blob/main/src/pages/tasks/TaskAssigneeSelectorModal.js) calls the [updateOptions](https://github.com/Expensify/App/blob/ed2c287daf6a1d59f4f43a9bfc84128597575730/src/pages/tasks/TaskAssigneeSelectorModal.js#L80) method and the screen shows the skeleton view(Please refer [here](https://github.com/Expensify/App/blob/ed2c287daf6a1d59f4f43a9bfc84128597575730/src/pages/tasks/TaskAssigneeSelectorModal.js#L198))


### What changes do you think we should make in order to solve the problem?

We can show the skeleton view until the `sections` is calculated. This is reasonable because we should show the skeleton while the data is being loaded. And I don't think filtered states are needed. They are all temporary values for calculating the `sections`.

So add the following changes
1. Add two states: `sections` and `isLoading`
```
    const [sections, setSections] = useState([]);
    const [isLoading, setIsLoading] = useState(true);
```

2. Move the `useMemo` body for calculating sections inside the `updateOptions` function and set state values(`sections` and `isLoading`)
```
        setSections(sectionsList);
        setIsLoading(false);
```

The final `updateOptions` function would look like
```
    const updateOptions = useCallback(() => {
        const {recentReports, personalDetails, userToInvite, currentUserOption} = OptionsListUtils.getNewChatOptions(
            props.reports,
            props.personalDetails,
            props.betas,
            searchValue.trim(),
            [],
            CONST.EXPENSIFY_EMAILS,
            false,
        );

        setHeaderMessage(OptionsListUtils.getHeaderMessage(recentReports?.length + personalDetails?.length !== 0 || currentUserOption, Boolean(userToInvite), searchValue));

        const sectionsList = [];
        let indexOffset = 0;

        if (currentUserOption) {
            sectionsList.push({
                title: props.translate('newTaskPage.assignMe'),
                data: [currentUserOption],
                shouldShow: true,
                indexOffset,
            });
            indexOffset += 1;
        }

        sectionsList.push({
            title: props.translate('common.recents'),
            data: recentReports,
            shouldShow: recentReports?.length > 0,
            indexOffset,
        });
        indexOffset += recentReports?.length;

        sectionsList.push({
            title: props.translate('common.contacts'),
            data: personalDetails,
            shouldShow: personalDetails?.length > 0,
            indexOffset,
        });
        indexOffset += personalDetails?.length;

        if (userToInvite) {
            sectionsList.push({
                data: [userToInvite],
                shouldShow: true,
                indexOffset,
            });
        }

        setSections(sectionsList);
        setIsLoading(false);
    }, [props, searchValue]);
```

3. Update the `shouldShowOptins` of the OptionsSelector [here](https://github.com/Expensify/App/blob/ed2c287daf6a1d59f4f43a9bfc84128597575730/src/pages/tasks/TaskAssigneeSelectorModal.js#L198)
```
        shouldShowOptions={!isLoading && didScreenTransitionEnd}
```

This works as expected

<details>
    <summary>Result</summary>


https://github.com/Expensify/App/assets/126839717/173f095c-d399-4108-86c7-3fc7dc4690ca


</details>

### What alternative solutions did you explore? (Optional)

**Comment by kadiealexander** [comment]
@sobitneupane please review the proposals above.

**Comment by sobitneupane** [comment]
Thanks for the update @pradeepmdk.

I am yet to review the updates. But some questions regarding your proposal, is applying `transitionSpec` enough to solve the issue? In `transitionSpec`, you have proposed to use fixed duration (300), will it work for large data?

You have added a solution [here](https://github.com/Expensify/App/issues/23563#issuecomment-1672445631). Is it part of your proposal or is it an alternative solution?

**Comment by s-alves10** [comment]
@sobitneupane 

What do you think on [my proposal](https://github.com/Expensify/App/issues/23563#issuecomment-1675796561)?

**Comment by pradeepmdk** [comment]
@sobitneupane 
yes this is another solution https://github.com/Expensify/App/issues/23563#issuecomment-1672445631 (this will work for larger data)

### Another solution 

in the `ScreenWrapper`  instead of the `transitionEnd` event we can use InteractionManager which will capture after all the animation end.

```javascript
 componentDidMount() {
   
        // this.unsubscribeTransitionEnd = this.props.navigation.addListener('transitionEnd', (event) => {
        //     // Prevent firing the prop callback when user is exiting the page.
        //     // console.log("transitionEnd", event);
        //     if (lodashGet(event, 'data.closing')) {
        //         return;
        //     }
        //     // this.setState({didScreenTransitionEnd: true});
        //     this.props.onEntryTransitionEnd();
        // });

        InteractionManager.runAfterInteractions(() => {
            console.log("complete intraction management");
            this.setState({didScreenTransitionEnd: true});
            this.props.onEntryTransitionEnd();
        });



        // We need to have this prop to remove keyboard before going away from the screen, to avoid previous screen look weird for a brief moment,
        // also we need to have generic control in future - to prevent closing keyboard for some rare cases in which beforeRemove has limitations
        // described here https://reactnavigation.org/docs/preventing-going-back/#limitations
        if (this.props.shouldDismissKeyboardBeforeClose) {
            this.beforeRemoveSubscription = this.props.navigation.addListener('beforeRemove', () => {
                if (!this.props.isKeyboardShown) {
                    return;
                }
                Keyboard.dismiss();
            });
        }
    }

``` 

**i think this is best solution for this.**

**Comment by pradeepmdk** [comment]
@sobitneupane 
the `transitionSpec` set (300ms) for delay the  trigger `transitionEnd` only. not for data loading based.
yes this is another solution https://github.com/Expensify/App/issues/23563#issuecomment-1672445631 (this will work for larger data)

### Another solution 

in the `ScreenWrapper`  instead of the `transitionEnd` event we can use InteractionManager which will capture after all the animation end.

```javascript
 componentDidMount() {
   
        // this.unsubscribeTransitionEnd = this.props.navigation.addListener('transitionEnd', (event) => {
        //     // Prevent firing the prop callback when user is exiting the page.
        //     // console.log("transitionEnd", event);
        //     if (lodashGet(event, 'data.closing')) {
        //         return;
        //     }
        //     // this.setState({didScreenTransitionEnd: true});
        //     this.props.onEntryTransitionEnd();
        // });

        InteractionManager.runAfterInteractions(() => {
            console.log("complete intraction management");
            this.setState({didScreenTransitionEnd: true});
            this.props.onEntryTransitionEnd();
        });

    }

``` 

** this work only for native. in web `transitionEnd` event only wokring correctly.**

**Comment by pradeepmdk** [comment]
@tienifr  this execution time differ is because of debounce.
so as I said when we navigate to a different stack navigator push will happen so at the time `transitionEnd` triggers immediately and background some other process is going on so at the time we are adding the value in the state so its take time.

but in the create time assignee we are in the same stack navigator `navigate` will happen  so some dealy happen on the time so with the time frame all background animation is completed. so after the dobounce will trigger so only debounce function will execute.

if you set it like this you can get the same execution time.
```javascript
        const debouncedSearch = _.debounce(updateOptions, 0);

``` 

@sobitneupane 
so I suggest one more solution is `InteractionManager`.

**Comment by pradeepmdk** [comment]
@tienifr  this execution time differ is because of debounce.
so as I said when we navigate to a different stack navigator push will happen so at the time `transitionEnd` triggers immediately and background some other process is going on so at the time we are adding the value in the state so its take time.

but in the create time assignee we are in the same stack navigator `navigate` will happen  so some dealy happen on the time so with the time frame all background animation is completed. so after the dobounce will trigger so only debounce function will execute.

if you set it like this you can get the same execution time.
```javascript
        const debouncedSearch = _.debounce(updateOptions, 0);

``` 

@sobitneupane 
so I suggest one more solution is `InteractionManager`.

**Comment by s-alves10** [comment]
@sobitneupane 

Did you check the RCA part of [my proposal](https://github.com/Expensify/App/issues/23563#issuecomment-1675796561)? I'd like to know your thought

**Comment by sobitneupane** [comment]
Thanks for the [detailed explanation](https://github.com/Expensify/App/issues/23563#issuecomment-1671797804) @pradeepmdk. It definitely made the root cause clear.

But I think instead of adding transitionSpec, it makes more sense to add `isLoading`. We already have a proposal with [similar solution](https://github.com/Expensify/App/issues/23563#issuecomment-1651223205).

**Comment by sobitneupane** [comment]
Thanks for the [proposal](https://github.com/Expensify/App/issues/23563#issuecomment-1675796561) @s-alves10.

Why do we need two separate states `section` and `isLoading`? I think `isLoding` will be enough to handle the situation.

**Comment by s-alves10** [comment]
@sobitneupane 

`sections` includes the data instead of old data: `sections`, `filteredRecentReports`, `filteredPersonalDetails`, `filteredUserToInvite`, and `filteredCurrentUserOption`.

**Comment by pradeepmdk** [comment]
@sobitneupane  Thank you.
can I get this task for finding the root cause clear. https://github.com/Expensify/App/issues/23563#issuecomment-1680620347

**Comment by pradeepmdk** [comment]
@sobitneupane  Thank you.

we need to consider this as well because other places also possibly have this issue 
when switching one stack navigator to another stack navigator 
https://github.com/Expensify/App/issues/23563#issuecomment-1679687110


can I get this task for finding the root cause clear.  https://github.com/Expensify/App/issues/23563#issuecomment-1680620347

**Comment by sobitneupane** [comment]
Thanks for the proposal everyone.

[Proposal](https://github.com/Expensify/App/issues/23563#issuecomment-1651223205) from @tienifr looks good to me.

🎀 👀 🎀 C+ reviewed

**Comment by sobitneupane** [comment]
I was also thinking that we should somehow reward @pradeepmdk for his RCA.

**Comment by Beamanator** [comment]
I def think it's fair to award for the RCA, what do you think @kadiealexander ? 

And I agree @tienifr 's proposal looks great 👍

**Comment by pradeepmdk** [comment]
@tienifr @sobitneupane @Beamanator @kadiealexander Thank you for everyone considering me as well. is there I also eligible to get this bonus amount?

because I am also suggesting with solution https://github.com/Expensify/App/issues/23563#issuecomment-1672445631 for this RCA.

**Comment by pradeepmdk** [comment]
@kadiealexander  @Beamanator  any updates

**Comment by sobitneupane** [comment]
**Regression Test Proposal**

1. Go to any task report
2. Click Assignee field
3. Verify contacts list shows immediately after skeleton and there is no blank page (empty screen) between skeleton and contact list.

Do we agree 👍 or 👎

**Comment by sobitneupane** [comment]
Requested payment on newDot.

https://github.com/Expensify/App/issues/23563#issuecomment-1686152546

**Comment by JmillsExpensify** [comment]
@kadiealexander Can you please provide a payment summary?

**Comment by kadiealexander** [comment]
Sorry team! I was looking out for Melvin's reminders. 

Payouts due:

- [ ] Issue Reporter: $250 @situchan
- [ ] Contributor: $2000 + $1000 bonus - @tienifr 
- [ ] Contributor+: $2000 + $1000 bonus - @sobitneupane

Eligible for 50% #urgency bonus? [Yes](https://github.com/Expensify/App/issues/23563#issuecomment-1686152546)

Upwork job is [here](https://www.upwork.com/jobs/~0163bd3cb559f30da0).

(cc @JmillsExpensify)

**Comment by kadiealexander** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [X] [@sobitneupane] Determine if we should create a regression test for this bug.
- [x] [@sobitneupane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [x] [@kadiealexander] Link the GH issue for creating/updating the regression test once above steps have been agreed upon: https://github.com/Expensify/Expensify/issues/315905

**Comment by pradeepmdk** [comment]
@kadiealexander  could you able to check this 
>I def think it's fair to award for the RCA, what do you think @kadiealexander ?


https://github.com/Expensify/App/issues/23563#issuecomment-1682111781
https://github.com/Expensify/App/issues/23563#issuecomment-1683828850
because I also have the solution https://github.com/Expensify/App/issues/23563#issuecomment-1672445631 for this RCA.

**Comment by kadiealexander** [comment]
@pradeepmdk we don't have a precedence for this, let me ask internally.

**Comment by pradeepmdk** [comment]
@kadiealexander Thank you.

I need some clarification on selecting the proposal.
I have corrected both RCA and the solution but I have multiple solutions (one of them now selected for fixing).
@tienifr has a solution only correctly.

where i am missing to selecting here?  

and due to https://github.com/Expensify/App/issues/23563#issuecomment-1655557258 being unclear of ROOT case only we are getting more proposals here

**Comment by kadiealexander** [comment]
I actually didn't pick the proposals, @sobitneupane could you please take a look at @pradeepmdk's question above?

**Comment by pradeepmdk** [comment]
https://github.com/Expensify/App/issues/23563#issuecomment-1682091999 
After selecting the proposal everybody including me.so  I thought I was also part of this. but now I am missing.

I agree that @tienifr posted the solution first. I posted after https://github.com/Expensify/App/issues/23563#issuecomment-1655557258 and increased the bounty.

but now I only have both RCA and solution first. 

😢

**Comment by sobitneupane** [comment]
Yup. We have [agreed](https://github.com/Expensify/App/issues/23563#issuecomment-1683828850) on awarding bounty to @pradeepmdk . I am not sure what is the right amount but I vaguely remember paying 25% in such cases.

cc: @kadiealexander

**Comment by pradeepmdk** [comment]
> Where i am missing in selecting my proposal here? https://github.com/Expensify/App/issues/23563#issuecomment-1714857271

Let me know what I am missing in selecting my proposal here

still confusing here 
@sobitneupane @Beamanator

**Comment by pradeepmdk** [comment]
any update

**Comment by kadiealexander** [comment]
We are discussing this [internally](https://expensify.slack.com/archives/C02NK2DQWUX/p1695107356179799) @pradeepmdk, thanks for your patience.

**Comment by pradeepmdk** [comment]
@kadiealexander thanks for the update.

**Comment by kadiealexander** [comment]
@pradeepmdk our team has reviewed the situation and we feel it's fair to award you 25% of the contributor bounty for your assistance in determining the root cause of this bug. As you were not materially involved in proposing or implementing the chosen solution, we do not agree that the full 100% would be fair here. 

Thanks again for your patience while we discussed this. Please apply [here](https://www.upwork.com/jobs/~0163bd3cb559f30da0) and we will get this paid out to you.

**Comment by kadiealexander** [comment]
@sobitneupane please don't forget about the first three points in the checklist [here](https://github.com/Expensify/App/issues/23563#issuecomment-1714554285). Thanks!

**Comment by kadiealexander** [comment]
Resurfacing this and updating it since it got buried.

Payouts due:

- [x] Issue Reporter: $250 @situchan
- [x] Contributor: $2000 + $1000 bonus - @tienifr 
- [ ] Contributor+: $2000 + $1000 bonus - @sobitneupane
- [ ] Bonus for RCA help: $500 - @pradeepmdk 

Eligible for 50% #urgency bonus? [Yes](https://github.com/Expensify/App/issues/23563#issuecomment-1686152546)

Upwork job is [here](https://www.upwork.com/jobs/~0163bd3cb559f30da0).

**Comment by pradeepmdk** [comment]
@kadiealexander  Thank you 

> As you were not materially involved in proposing or implementing the chosen solution, 

I proposed the solution here  https://github.com/Expensify/App/issues/23563#issuecomment-1672445631 for chosen one.

**Comment by kadiealexander** [comment]
@pradeepmdk thank you for your additional information, however our decision here is final. I have sent you a contract in Upwork for the payment, please accept and I will ensure you are paid quickly. 😊

**Comment by pradeepmdk** [comment]
@kadiealexander I agree your decision is final.  but it is still weird here because for no reason my RCA and solution were ignored. but no one has given me a solution as to why the proposal was not selected here.

**Comment by kadiealexander** [comment]
For clarity, @sobitneupane explained [here](https://github.com/Expensify/App/issues/23563#issuecomment-1680620347) why we preferred @tienifr's solution, even though your RCA was valuable here. I appreciate this was a competitive issue with lots of similar proposals, but ultimately our team reserves the right to choose the solution we believe is best.

**Comment by JmillsExpensify** [comment]
$3,000 payment for @sobitneupane approved based on BZ summary.
